engine/runtime/src/test/java/org/enso/compiler/ExecCompilerTest.java

```diff
@@ -174,6 +199,15 @@ public void chainedSyntaxOperator() throws Exception {
     assertEquals("10 % 3 is one", 1, result.asInt());
   }
 
+  @Test
+  public void inlineReturnSignatureWithoutArguments() throws Exception {
+    var module = ctx.eval("enso", """
+    the_number -> Integer = 23
+    """);
+    var result = module.invokeMember("eval_expression", "the_number");
+    assertEquals("Function-return syntax can be used with 0 arguments", 23, result.asInt());
+  }
```

Comment:
I think we should also get a test when there is more than 0 arguments 🙂 

I'd be happy to add a test case, may I push to your branch?